Base/Python/slicer/util.py

```diff
@@ -2306,75 +2306,103 @@ def delayDisplay(message, autoCloseMsec=1000):
   return result == qt.QMessageBox.Ok
 
 def confirmYesNoDisplay(text, windowTitle=None, parent=None, **kwargs):
-  """Display an confirmation popup. Return if confirmed with Yes."""
-  import qt, slicer
+  """Display a confirmation popup. Return if confirmed with Yes.
+
+  When the application is running in testing mode (``slicer.app.testingEnabled() == True``),
+  the popup is skipped and True ("Yes") is returned, with a message being logged to indicate this.
+  """
+  import qt, slicer, logging
   if not windowTitle:
     windowTitle = slicer.app.applicationName + " confirmation"
-  result = messageBox(text, parent=parent, windowTitle=windowTitle, icon=qt.QMessageBox.Question,
-                       standardButtons=qt.QMessageBox.Yes | qt.QMessageBox.No, **kwargs)
+  result = _messageDisplay(logging.INFO, text, True, parent=parent, windowTitle=windowTitle, icon=qt.QMessageBox.Question,
+                            standardButtons=qt.QMessageBox.Yes | qt.QMessageBox.No, **kwargs)
   return result == qt.QMessageBox.Yes
 
 def confirmRetryCloseDisplay(text, windowTitle=None, parent=None, **kwargs):
-  """Display an confirmation popup. Return if confirmed with Retry."""
-  import qt, slicer
-  if not windowTitle:
-    windowTitle = slicer.app.applicationName + " error"
-  result = messageBox(text, parent=parent, windowTitle=windowTitle, icon=qt.QMessageBox.Critical,
-                       standardButtons=qt.QMessageBox.Retry | qt.QMessageBox.Close, **kwargs)
+  """Display an error popup asking whether to retry, logging the text at error level.
+  Return if confirmed with Retry.
+
+  When the application is running in testing mode (``slicer.app.testingEnabled() == True``),
+  the popup is skipped and False ("Close") is returned, with a message being logged to indicate this.
+  """
+  import qt, logging
+  result = _messageDisplay(logging.ERROR, text, False, parent=parent, windowTitle=windowTitle,
+                            icon=qt.QMessageBox.Critical, standardButtons=qt.QMessageBox.Retry | qt.QMessageBox.Close, **kwargs)
   return result == qt.QMessageBox.Retry
 
+def _messageDisplay(logLevel, text, testingReturnValue, mainWindowNeeded=False, parent=None, windowTitle=None, **kwargs):
+  """Displays a messagebox and logs message text; knows what to do in testing mode.
+
+  Args:
+    logLevel: The level at which to log text, e.g. logging.INFO, logging.ERROR
+    text (str): Message text
+    testingReturnValue: When the application is in testing mode, this value is returned instead of raising the message box.
+    mainWindowNeeded (bool): If True then the message box will not be raised if there is no mainWindow, but the text is still logged.
+    parent (QWidget): The message box parent; by default it is set to main window by slicer.util.messageBox
+    windowTitle (str): Window title; defaults to a generic title based on log level.
+    kwargs: passed to slicer.util.messageBox
```

Comment:
```suggestion
  :param logLevel: The level at which to log text, e.g. ``logging.INFO``, ``logging.ERROR``
  :param text: Message text
  :type text: str
  :param testingReturnValue: When the application is in testing mode, this value is returned instead of raising the message box.
  :param mainWindowNeeded: If True then the message box will not be raised if there is no mainWindow, but the text is still logged.
  :type mainWindowNeeded: bool, optional
  :param parent: The message box parent; by default it is set to main window by slicer.util.messageBox
  :type parent: QWidget, optional
  :param windowTitle: Window title; defaults to a generic title based on log level.
  :type windowTitle: str, optional
  :param kwargs: passed to :func:`messageBox`
```

For sake of consistency with existing docstring  style in this file